### ISSUE TITLE
Support BSON Decimal128 type

### ISSUE THREAD

**Issue by jyemin** [michel-kraemer/bson4jackson#74]
MongoDB 3.4 added a new BSON type: [Decimal128](https://github.com/mongodb/specifications/blob/master/source/bson-decimal128/decimal128.rst)

Java support came in version 3.4 of the driver, and was not backported to 2.x.  References:

* Value object: https://github.com/mongodb/mongo-java-driver/blob/3.4.x/bson/src/main/org/bson/types/Decimal128.java
* Encoder: https://github.com/mongodb/mongo-java-driver/blob/3.4.x/bson/src/main/org/bson/BsonBinaryWriter.java#L211-L216
* Decoder: https://github.com/mongodb/mongo-java-driver/blob/3.4.x/bson/src/main/org/bson/BsonBinaryReader.java#L197-L201

**Comment by nsalhaji** [comment]
Hi guys,

Any plan to add support of Decimal128 on bson4jackson ?

```
Caused by: com.fasterxml.jackson.core.JsonParseException: Unknown element type 19
 at [Source: de.undercouch.bson4jackson.io.LittleEndianInputStream@11c88cca; pos: 137]
	at de.undercouch.bson4jackson.BsonParser.nextToken(BsonParser.java:381)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:271)
```

**Comment by zigzago** [comment]
@nsalhaji  this [fork of bson4jackson](https://search.maven.org/#artifactdetails%7Corg.litote%7Cbson4jackson-fork%7C2.9.0%7Cjar) supports Decimal128 (you need to enable the feature BsonGenerator.Feature.WRITE_BIGDECIMALS_AS_DECIMAL128).

If you prefer to create your own fork, here is the commit: https://github.com/Litote/bson4jackson/commit/323e35b18b1dcf156bb7b52d33178323de1d88a8

HTH

**Comment by nsalhaji** [comment]
Hi @zigzago,

Thanks

**Comment by michel-kraemer** [comment]
@zigzago That's very cool! Fancy making a pull request?

**Comment by zigzago** [comment]
@michel-kraemer Sure! Will do in the days ahead! :)

**Comment by nizarsalhaji** [comment]
Super.
@michel-kraemer any plan for the next release ?

Thanks guys.

**Comment by michel-kraemer** [comment]
Sure. I need to release a new version anyhow because of #76. I'll also go through the other issues before that. My plan is to release it by the end of next week.

**Comment by nizarsalhaji** [comment]
Thanks.

**Comment by michel-kraemer** [comment]
I just merged #77 and will release bson4jackson 2.9.0 either later today or tomorrow.

**Comment by nizarsalhaji** [comment]
Super. Thanks.
